Software 0.33 release notes (#2611)
astronomer/docs: software/release-notes.md

<!--- Version-specific -->

0.33 is the latest stable version of Astronomer Software, while 0.32 remains the latest long-term support (LTS) release. To upgrade to 0.33, see [Upgrade Astronomer](upgrade-astronomer.md). For more information about Software release channels, see [Release and lifecycle policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI release notes](https://docs.astronomer.io/astro/cli/release-notes).

## 0.33.0

Release date: August 3, 2023

### Automatic PGBouncer connection scaling

> Astronomer Software now automatically scales the size of PGBouncer connection pools based on the connection usage of your current Airflow components. This should result in improved performance, scalability, and utilization of database connections across all Deployments. 

Reviewer: `based on the connection usage of your current Airflow components`
I think this implies that we are measuring the connection usage of their currently deployed components, which we are not. I find it little misleading. Thoughts? @vishwas-astro @rishkarajgi @jwitz @pgvishnuram

Author: How is it measured @rujhan-arora-astronomer ? It's very possible I misunderstood https://github.com/astronomer/issues/issues/5618, I assumed we dynamically scaled based on a component's connection usage at a given moment. Are we instead scaling for what a component would generically use for all Deployments?

Reviewer 2: ```suggestion
Astronomer Software now automatically scales the size of PGBouncer connection pools based on your Airflow component counts and Airflow configuration, instead of solely based on total AU. This should result in improved performance, scalability, and utilization of database connections across all Deployments. 
```

Something like this?

### Additional improvements

- Added support for [Kubernetes 1.27](https://kubernetes.io/blog/2023/04/11/kubernetes-v1-27-release/).
- The Workspace **Deployments** page is now paginated in the Astronomer UI.
- You no longer have to set `elasticsearch.curator.age.timestring` when you configure a custom indexing pattern for [Vector logging sidecars](export-task-logs.md#export-logs-using-container-sidecars). The only required value is now `astronomer.houston.config.deployments.helm.loggingSidecar.indexPattern`.

> - You can now set `deployments.canCreateDeploymentFromUI` to false to prevent all users from creating new Deployments through the Astronomer UI. 

Reviewer 3: This is now `canUpsertDeploymentFromUI` and it impacts three views. 

- Create Deployment View
- Update Deployment View
- Deployment Variables View

The flag applies only to non system admins

> - You can now set `deployments.canCreateDeploymentFromUI` to false to prevent all users from creating new Deployments through the Astronomer UI. 

Author: ```suggestion
- You can now set `deployments.canUpsertDeploymentFromUI: false` to prevent all users besides System Admins from updating Deployments and environment variables through the Astronomer UI. 
```

### Bug fixes

- Fixed an issue where configurations in `astronomer.houston.config.deployments.components` applied only to new Deployments made after the configuration was set. 
- Fixed an issue where a Workspace-level service account would improperly inherit lesser permissions for Deployments it was added to.

> - Fixed an issue where the Astronomer UI would still show a **Delete** button for Teams for roles that did not have the `system.teams.remove` permission.

Reviewer 3: This fix actually now shows an appropriate error message we did not remove nor disable for this case scenario

> - Fixed an issue where the Astronomer UI would still show a **Delete** button for Teams for roles that did not have the `system.teams.remove` permission.

Reviewer 3: This fix actually now shows an appropriate error message when the delete is not successful (as against the success message it was showing even though it was not deleted). We did not remove nor disable the button for this case scenario.

- Fixed an issue where you could get a 500 internal server error from the Airflow UI when switching between pages for a DAG.
- Fixed an issue where you couldn't set `properties.email` using the `upsertDeployment` mutation.

> - Fixed an issue where the Astronomer UI would freeze if a System Admin user viewed service accounts without the appropriate permissions. 

Reviewer 3: This fix is not limited to the system admins. You could reword to this

```suggestion
- Fixed an issue where the Astronomer UI would only show loading dots instead of Access denied when a user without the appropriate permissions views service accounts. 
```

- The **Extra Capacity** field in the Astronomer UI now shows up to 6 digits of AU.
- You no longer have to set `elasticsearch.curator.age.timestring` when you configure a custom indexing pattern for [Vector logging sidecars](export-task-logs.md#export-logs-using-container-sidecars). The only required value is now `astronomer.houston.config.deployments.helm.loggingSidecar.indexPattern`.
- When you create or update a Deployment and select a Runtime version, the Astronomer UI now shows only the latest supported Astro Runtime patch for each supported Astro Runtime major version.
- You can now set `deployments.canUpsertDeploymentFromUI: false` to prevent all users besides System Admins from updating Deployments and environment variables through the Astronomer UI.
- You can now [overprovision](cluster-resource-provisioning.md) the `triggerer-log-groomer` component.

### Bug fixes

- Fixed an issue where configurations in `astronomer.houston.config.deployments.components` applied only to Deployments that were created after the configuration was set. 
- Fixed an issue where a Workspace-level service account would improperly inherit lesser permissions for Deployments it was added to.

> - Fixed an issue where the Astronomer UI would still show a **Delete** button for Teams for roles that did not have the `system.teams.remove` permission.

Author: ```suggestion
- The Astronomer UI now shows an error if you click the **Delete** button for Teams and you don't have the `system.teams.remove` permission.
```

<!--- Version-specific -->

0.33 is the latest stable version of Astronomer Software, while 0.32 remains the latest long-term support (LTS) release. To upgrade to 0.33, see [Upgrade Astronomer](upgrade-astronomer.md). For more information about Software release channels, see [Release and lifecycle policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI release notes](https://docs.astronomer.io/astro/cli/release-notes).

## 0.33.0

Release date: August 3, 2023

### Automatic PGBouncer connection scaling

> Astronomer Software now automatically scales the size of PGBouncer connection pools based on your Airflow component counts and Airflow configuration, instead of solely based on total AU. This should result in improved performance, scalability, and utilization of database connections across all Deployments. 

Reviewer 4: ```suggestion
Astronomer Software now automatically scales the size of PGBouncer connection pools based on your Airflow component counts and Airflow configuration, instead of solely based on total AU. This improves performance, scalability, and utilization of database connections across all Deployments. 
```

### Additional improvements

- You can now disable Airflow and platform alerts on the Prometheus alerts dashboard by setting `prometheus.defaultAlerts.airflow.enabled` and `prometheus.defaultAlerts.airflow.enabled` to `false` in your Prometheus Helm chart. If you disable these alerts, you can still add back specific alerts or configure custom alerts using `prometheus.defaultAlerts.additionalAlerts`. See [Create custom alerts](platform-alerts.md#create-custom-alerts).
- Added support for [Kubernetes 1.27](https://kubernetes.io/blog/2023/04/11/kubernetes-v1-27-release/).
- The Workspace **Deployments** page is now paginated in the Astronomer UI.
- The **Extra Capacity** field in the Astronomer UI now shows up to 6 digits of AU.

> - You no longer have to set `elasticsearch.curator.age.timestring` when you configure a custom indexing pattern for [Vector logging sidecars](export-task-logs.md#export-logs-using-container-sidecars). The only required value is now `astronomer.houston.config.deployments.helm.loggingSidecar.indexPattern`.

Reviewer 5: astronomer.houston.config.deployments.helm.loggingSidecar.indexPattern ->   global.loggingSidecar.indexPattern 

this is the correct place to add

- When you create or update a Deployment and select a Runtime version, the Astronomer UI now shows only the latest supported Astro Runtime patch for each supported Astro Runtime major version.
- You can now set `deployments.canUpsertDeploymentFromUI: false` to prevent all users besides System Admins from updating Deployments and environment variables through the Astronomer UI.
- You can now [overprovision](cluster-resource-provisioning.md) the `triggerer-log-groomer` component.

### Bug fixes

> 

Author: ```suggestion

- Fixed an issue where a Deployment using Runtime 8 or earlier with the Celery executor would show as healthy in the Software UI even when workers were unavailable.
```

<!--- Version-specific -->

0.33 is the latest stable version of Astronomer Software, while 0.32 remains the latest long-term support (LTS) release. To upgrade to 0.33, see [Upgrade Astronomer](upgrade-astronomer.md). For more information about Software release channels, see [Release and lifecycle policies](release-lifecycle-policy.md). To read release notes specifically for the Astro CLI, see [Astro CLI release notes](https://docs.astronomer.io/astro/cli/release-notes).

## 0.33.0

Release date: August 3, 2023

### Automatic PGBouncer connection scaling

> Astronomer Software now automatically scales the size of PGBouncer connection pools based on your Airflow component counts and Airflow configuration, instead of solely based on total AU. This improves performance, scalability, and utilization of database connections across all Deployments. 

Reviewer: @jwitz 
Customer needs to set `deployments.pgBouncerResourceCalculationStrategy` key in the config to `airflowStratV2`, in order to use this feature 
cc: @rishkarajgi @danielhoherd

Reviewer 6: https://github.com/astronomer/docs/pull/2817
